src/PaprController.sol

```diff
@@ -283,45 +283,31 @@ contract PaprController is
         address sendTo,
         ReservoirOracleUnderwriter.OracleInfo calldata oracleInfo
     ) external override {
-        uint256 price = _purchaseNFTAndUpdateVaultIfNeeded(auction, maxPrice, sendTo);
-        --_vaultInfo[auction.nftOwner][auction.auctionAssetContract].auctionCount;
+        // This function is almost the same as NFTEDA._purchaseNFT
+        // however we inline so we can receive payment first
+        // and execute logic before sending NFT
+        uint256 id;
```

Comment:
is it gas beneficial to init these vars before assigning them?

Reply:
no was just a product of copying code from NFTEDA where some of these were already inited by the functions return values